src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,76 @@
-export const Pagination = () => {};
+import React from 'react';
+import classNames from 'classnames';
+import { getNumbers } from '../../utils';
+
+type Props = {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: (pageNumber: number) => void
+};
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const pages = getNumbers(1, Math.ceil(total / perPage));
+
+  return (
+    <ul className="pagination">
+      <li className={classNames(
+        'page-item',
+        { disabled: currentPage === 1 },
+      )}
+      >
+        <a
+          data-cy="prevLink"
+          className="page-link"
+          href="#prev"
+          aria-disabled={currentPage === 1}
+          onClick={() => (currentPage !== 1
+            ? onPageChange(currentPage - 1)
+            : null)}
+        >
+          «
+        </a>
+      </li>
+      {pages.map(page => (
+        <li
+          key={page}
+          className={classNames(
+            'page-item',
+            { active: page === currentPage },
+          )}
+        >
+          <a
+            data-cy="pageLink"
+            className="page-link"
+            href={`#${page}`}
+            onClick={() => onPageChange(page)}
+          >
+            {page}
+          </a>
+        </li>
+      ))}
+      <li className={classNames(
+        'page-item',
+        { disabled: currentPage === pages.length },
```

Comment:
you make one check several times, will be better to create separate boolean variable and use it where you need it